### ISSUE TITLE
[HF][Optimum] Weights Neff separation not implemented for PyTorch 2.X in Neuron SDK 2.17

### ISSUE THREAD

**Issue by JingyaHuang** [aws-neuron/aws-neuron-sdk#846]
Hi team, Optimum Neuron team is going to transit from default PyTorch 1.13 support to PyTorch 2.1 support, and we noticed that the weights / neff separation support while tracing the model and the `replace_weights()` support is only implemented for torch_neuronx 1.13.X but not 2.1.X(`torch_neuronx==2.1.1.2.0.1b0`).

Given that this feature is predominant for the wip caching system of traced models, we wonder if this feature would be implemented for Torch 2.X during the Neuron SDK 2.18 release.

```
***** Compiling encoder *****
Using Neuron: --auto-cast matmul
Using Neuron: --auto-cast-type bf16
/home/ubuntu/pyvenv/aws_neuron_venv_torch_2.1/lib/python3.8/site-packages/torch_neuronx/xla_impl/trace.py:368: UserWarning: Unexpected keyword arguments: ['inline_weights_to_neff']
```

**Comment by mrnikwaws** [comment]
Hi @JingyaHuang,

Thanks for reporting the issue.  I can confirm that the fix for torch_neuronx 2.1.X is in our release pipeline and will appear in an upcoming release.

We'll post back here when we release and close the issue.

**Comment by JingyaHuang** [comment]
Confirm that the feature is implemented in Neuron SDK 2.18 for both pt1.13 and pt2.1, closing the issue. Thanks the team for helping!
